src/views/Home.jsx

```diff
@@ -9,13 +9,13 @@ export function Home({ data, setListPath, userEmail, userId }) {
 	const [notificationMessage, setNotificationMessage] = useState('');
 	const navigate = useNavigate();
 	const previousDataRef = useRef([]);
-	previousDataRef.current = data;
 
 	useEffect(() => {
 		const newestList = data[data.length - 1];
 		setListPath(newestList?.path);
 
 		if (previousDataRef.current.length === 0) {
+			previousDataRef.current = data;
 			return;
```

Comment:
Just a couple more notes.

You don't need this return because at this point the `ref === data` and the second `if` clause will be false so it won't redirect. 

You don't need the return cleanup function because every time we create a list we route away from the Home component. Every time we route back to the Home component the ref state is set back to `[ ]` and the first `if` block will run.

Lines 14 & 15 should go into the 2nd `if` block so we're not running the `useStateWithStorage` hook on the first render. We don't need a new list path until the data changes.